### PR TITLE
Add path param support

### PR DESCRIPTION
**Pull request by jessealva**

### CODE REVIEW

**Comment by jessealva** [comment]
@mdeuser could you review this real quick? its a change to enable parameters as children of Paths

**Comment by dubeejw** [comment]
@jessealva, can we unit test this?

File: whisk/api.go [hunk 1]
```diff
@@ -146,13 +146,51 @@ type ApiSwagger struct {
     SwaggerName     string    `json:"swagger,omitempty"`
     BasePath        string    `json:"basePath,omitempty"`
     Info            *ApiSwaggerInfo `json:"info,omitempty"`
-    Paths           map[string]map[string]*ApiSwaggerOperation `json:"paths,omitempty"`  // Paths["/a/path"]["get"] -> a generic object
+    Paths           map[string]*ApiSwaggerPath `json:"paths,omitempty"`  // Paths["/a/path"]["get"] -> a generic object
```
Comment (mdeuser) [inline whisk/api.go]: i think the comment is incorrect.  should be something like
```// Paths["/a/path"] -> a path (ApiSwaggerPath) object```

Comment (jessealva) [reply whisk/api.go]: Removing the comment as I think its no longer required as it was there when the object was a map of maps.

**Comment by jessealva** [comment]
@dubeejw So unless I'm mistaken there is no testing infrastructure setup already?  I'm afraid if I attempt to test this then it will prevent me from actually getting this feature in this month.  Looks like we already have a feature request for this #14 .  Could we just do this testing at that point?

**Comment by dubeejw** [comment]
@jessealva, looks great! Do you have tests for this at least in incubator-openwhisk-cli?

**Pull request merged**